docs/auditors/deprecatedapis.md

```diff
@@ -0,0 +1,96 @@
+# Kubernetes Deprecated API Auditor (deprecatedapis)
+
+Finds any resource defined with adeprecated API version.
+
+## General Usage
+
+```
+kubeaudit deprecatedapis [flags]
+```
+
+### Flags
+| Short   | Long                   | Description                                   | Default             |
+| :------ | :--------------------- | :-------------------------------------------- | :------------------ |
+|         | --current-k8s-version  | Kubernetes current version                    |                     |
+|         | --targeted-k8s-version | Kubernetes version to migrate to              |                     |
```

Comment:
I'm a bit unclear why there is a flag for both current and target version. Maybe I'm misunderstanding the use case, but to me it sounds easier to reason about if we had just the target version:

- `ERROR` if the resource API is removed in the target version
- `WARN` if the resource API is deprecated in the target version
- `INFO` if the resource API is not deprecated in the target version but will be deprecated in a future version

We could default to the current version in local and cluster mode since kubeaudit should be able to look that up automatically?

Reply:
If we specify a `current version` or this information is retrieved from the cluster without a targeted version we can have the following cases:
- `ERROR` if the resource API is removed in the current version
- `WARN` if the resource API is deprecated in the current version
- `INFO` if the resource API is not deprecated in the current version but will be deprecated in a future version

If we use a targeted version the message level takes precedence over the current version.

But maybe we can remove this option and only use the current version in the local or in cluster mode 🤔

Reply:
Ah I see from your flag description

```
--targeted-k8s-version string   Kubernetes version to migrate to (eg 1.24)
```
The use case for this auditor is for doing Kubernetes version migrations?

Based on the description of the auditor "This command determines which resource is defined with a deprecated API version." I was thinking of "target version" as "check Kubernetes resources against this version", hence it didn't make sense to me to have both "current" and "target" versions. Did you have a specific workflow in mind where you would benefit from passing both current and target versions (eg. running against a live cluster as a one-off before doing a migration)? If so, I think we should document it (eg. as an example) to make the use of the flags more clear. Nothing (other than the target flag description) says anything about migrations.

Reply:
Many APIs have been declared deprecated since the 1.16 but are removed from the version 1.22.  If the current version is before the 1.16 we can keep the API as it is but  if it is not the case, it is better to migrate the API as soon as possible. That is why a warning is generated from the version 1.16.

But as currently `kubeaudit` does not retrieve  the current version from the cluster we can remove this option in a first time.